launcher/FileSystem.cpp

```diff
@@ -279,11 +631,60 @@ QString PathCombine(const QString& path1, const QString& path2, const QString& p
     return PathCombine(PathCombine(path1, path2, path3), path4);
 }
 
-QString AbsolutePath(QString path)
+QString AbsolutePath(const QString& path)
 {
     return QFileInfo(path).absolutePath();
 }
 
+int PathDepth(const QString& path)
```

Comment:
It's a good thought but without splitting it into parts there isn't a clean way to detect the difference between `".."` and `"."` as a path part. the (now renamed) `pathDepth` function will not always be given a absolute path so not all dot components will be removed. by `QDir::cleanPath`

Reply:
The relative path issue could be solved by doing something like this before calling `QDir::cleanPath` (would need to make `path` a copy), if i understand it correctly:
```cpp
if (QDir::isRelativePath(path))
    path = PathCombine(QDir::currentPath(), path);
```

Reply:
It would resolve the path but it would change the behavior of the function.
the path depth I need it to return is the depth from the top of the path given.